public/js/app/components/Table.jsx

```diff
@@ -1,28 +1,33 @@
 import React from 'react'
 
-const renderRow = (row) => (
-  <tr key={row[0]}>
-    {row.map((cell, index) => (
-      <td key={index}>{typeof cell === 'function' ? cell() : cell}</td>
-    ))}
-  </tr>
-)
+const renderRow = (row, rowIndex) => {
+  return (
+    //TODO: unique ID for key
+    <tr key={rowIndex}>
+      {row.map((cell, cellIndex) => (
+        <td key={cellIndex}>{typeof cell === 'function' ? cell() : cell}</td>
+      ))}
+    </tr>
+  )
+}
 
-const Table = ({ headings, rows, tableClasses }) => (
-  <>
-    <table className={tableClasses.join(' ')}>
-      <thead>
-        <tr>
-          {headings.map((h) => (
-            <th key={h} scope="col">
-              {h}
-            </th>
-          ))}
-        </tr>
-      </thead>
-      <tbody>{rows.map((r) => renderRow(r))}</tbody>
-    </table>
-  </>
-)
+const Table = ({ headings, rows, tableClasses }) => {
+  return (
+    <>
+      <table className={tableClasses.join(' ')}>
+        <thead>
+          <tr>
+            {headings.map((heading, index) => (
+              <th key={index} scope="col" className="semester-column">
+                {heading}
+              </th>
+            ))}
+          </tr>
+        </thead>
+        <tbody>{rows.map((row, index) => renderRow(row, index))}</tbody>
```

Comment:
On row-level maybe we can find a good ID in the data, that we can send along with the row.